app/controllers/api/stateless/braintree/payment_methods_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+module Api
+  module Stateless
+    module Braintree
+      class PaymentMethodsController < StatelessController
+        before_filter :authenticate_request!
+
+        def index
+          @payment_methods = PaymentHelper::Braintree.payment_methods_for_member(@current_member)
+        end
+
+        def destroy
+          @payment_method = PaymentHelper::Braintree.payment_method_for_member(member: @current_member, id: params[:id])
+
+          result = begin
+                     ::Braintree::PaymentMethod.delete(@payment_method.token)
+                   rescue ::Braintree::NotFoundError
+                     @payment_method.destroy
+                   end
+
+          @payment_method.destroy if result.success?
```

Comment:
If the record for some reason does not exist on Braintree (e.g. it's already been deleted), we delete the local record, and if that's successful, then the record is deleted in both places, as expected, and the result is successful. If the local record lingers, we didn't get what we wanted. The logic you list here is not very different from how subscription cancellations are handled - and I'm planning to update the payment method controller to use the same pattern - the only difference is that the local record gets destroyed explicitly under the BT:NotFound rescue block and the data returned to the client looks a bit different in these two cases.